compiler/rustc_hir_analysis/src/coherence/builtin.rs

```diff
@@ -94,7 +95,14 @@ fn visit_implementation_of_copy(tcx: TyCtxt<'_>, impl_did: LocalDefId) {
             let mut errors: BTreeMap<_, Vec<_>> = Default::default();
             let mut bounds = vec![];
 
+            let mut seen_tys = FxHashSet::default();
+
             for (field, ty, reason) in fields {
+                // Only report an error once per type.
```

Comment:
so if more than one field isn't copy, it won't report all fields? am i understanding that right?

Reply:
It does not highlight subsequent fields with identical types.

Reply:
oh once per *field type*, got it.

Reply:
This is actually unrelated to the example code you gave me, and was something that @fee1-dead had pointed out somewhere else a few days ago. It just came to mind when I was making the change.

Reply:
I think it was from @WaffleLapkin? Anyways thanks for working on this!

Reply:
oops, just remember seeing a screenshot of a really bad error message somewhere :rofl:

Reply:
The screenshot in question:
![2023-03-03_15-05](https://user-images.githubusercontent.com/38225716/223679617-23d35134-db07-4f65-96c0-928abc377d0f.png)

And ye, thanks for fixing this, errs <3